examples/adc_buffer.rs

```diff
@@ -7,8 +7,12 @@ use libtock::console::Console;
 use libtock::result::TockResult;
 use libtock::syscalls;
 
+fn main() {
+    let _ = main_impl();
+}
+
 /// Reads a 128 byte sample into a buffer and prints the first value to the console.
-async fn main() -> TockResult<()> {
+fn main_impl() -> TockResult<()> {
```

Comment:
Why does this have to be treated differently? I would be okay with an async function containing only synchronuous code.

Reply:
I cleaned this up a bit. I'm keeping it synchronous because making it asynchronous is not useful.